src/main/java/ch/njol/skript/util/EnumUtils.java

```diff
@@ -28,90 +28,93 @@
 import java.util.HashMap;
 import java.util.Locale;
 
+/**
+ * A language utility class to be used for easily handling language values representing an Enum.
+ * @param <E> Generic representing the Enum.
+ * @see ch.njol.skript.classes.EnumClassInfo
+ */
 public final class EnumUtils<E extends Enum<E>> {
 	
-	private final Class<E> c;
+	private final Class<E> enumClass;
 	private final String languageNode;
 
+	@SuppressWarnings("NotNullFieldNotInitialized") // initialized in constructor's refresh() call
 	private String[] names;
 	private final HashMap<String, E> parseMap = new HashMap<>();
 	
-	public EnumUtils(Class<E> c, String languageNode) {
-		assert c != null && c.isEnum() : c;
-		assert languageNode != null && !languageNode.isEmpty() && !languageNode.endsWith(".") : languageNode;
+	public EnumUtils(Class<E> enumClass, String languageNode) {
+		assert enumClass.isEnum() : enumClass;
+		assert !languageNode.isEmpty() && !languageNode.endsWith(".") : languageNode;
 		
-		this.c = c;
+		this.enumClass = enumClass;
 		this.languageNode = languageNode;
 
-		names = new String[c.getEnumConstants().length];
+		refresh();
 		
-		Language.addListener(() -> validate(true));
+		Language.addListener(this::refresh);
 	}
 	
 	/**
-	 * Updates the names if the language has changed or the enum was modified (using reflection).
+	 * Refreshes the representation of this Enum based on the currently stored language entries.
 	 */
-	void validate(boolean force) {
-		boolean update = force;
-
-		E[] constants = c.getEnumConstants();
+	void refresh() {
+		E[] constants = enumClass.getEnumConstants();
+		names = new String[enumClass.getEnumConstants().length];
+		parseMap.clear();
+		for (E constant : constants) {
+			String key = languageNode + "." + constant.name();
+			int ordinal = constant.ordinal();
 
-		if (constants.length != names.length) { // Simple check
-			names = new String[constants.length];
-			update = true;
-		} else { // Deeper check
-			for (E constant : constants) {
-				if (!parseMap.containsValue(constant)) { // A new value was added to the enum
-					update = true;
-					break;
+			String[] options = Language.getList(key);
+			for (String option : options) {
+				option = option.toLowerCase(Locale.ENGLISH);
+				if (options.length == 1 && option.equals(key.toLowerCase(Locale.ENGLISH))) {
+					Skript.debug("Missing lang enum constant for '" + key + "'");
+					continue;
```

Comment:
Rather than just continue, can we make this register a default key like `typeName.constant`?  ex: `attributetype.generic_max_absorbtion`. This allows usage of new enums until skript properly adds them into the file. It'll like default aliases.

Reply:
I discussed this with the team and for now we are not going to implement this. These kinds of default values work with aliases since they are based directly on the Minecraft ID, but we want to avoid encouraging the usage of temporary enum constants that may not end up being valid in an actual release.

Reply:
Alright, sad, but I see the cause